### PR TITLE
implemented the persistent state across the app

### PR DESCRIPTION
**Pull request by alexander-dev-hub**

### CODE REVIEW

File: components/StateContext/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+import { createContext } from 'react';
+
+const StateContext = createContext();
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply components/StateContext/index.js]: Naming solved.

File: components/SwitchWithLabel/index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import Toggle from 'react-toggle'; // TODO: update -> deprecated lifecyle usage
 const SwitchWithLabel = ({ label, ...rest }) => {
   return (
     <div className='switch-with-label'>
-      <Toggle {...rest}/>
+      <Toggle {...rest} />
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply components/SwitchWithLabel/index.js]: thank you.

File: pages/AdaptivePostList/index.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 
-import { Suspense, Fragment, useState } from 'react';
+import { Suspense, Fragment, useContext } from 'react';
 
+import StateContext from '../../components/StateContext';
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/AdaptivePostList/index.js]: Solved.

File: pages/AdaptivePostList/index.js [hunk 4]
```diff
@@ -10,17 +11,12 @@ import { useMemoryStatus } from '../../utils/hooks';
...
   const memoryStatus = useMemoryStatus();
-
-  const enableManualAnimationHandler = flag => {
-    setManualEnabled(flag);
-  };
-
-  const toggleAnimationHandler = event => {
-    setAnimationOn(event.target.checked);
-  };
+  const {
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/AdaptivePostList/index.js]: solved

File: pages/AdaptivePostList/index.js [hunk 5]
```diff
@@ -42,6 +38,7 @@ const AdaptivePostList = () => {
     <Fragment>
       <Nav 
         manualEnabled={manualEnabled}
+        isAnimationOn={isAnimationOn}
         enableManualAnimation={enableManualAnimationHandler}
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/AdaptivePostList/index.js]: solved it by using AnimationEmulationContext in Nav component.

File: pages/AnimationPost/[index].js [hunk 6]
```diff
@@ -1,11 +1,10 @@
 import { useEffect } from 'react';
 
 import Layout from '../../components/layout';
-
+import PostInfo from '../../components/PostInfo';
 import Link from 'next/link';
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/AnimationPost/[index].js]: Solved.

File: pages/AnimationPost/[index].js [hunk 7]
```diff
@@ -1,11 +1,10 @@
...
 import posts from '../../data/posts';
 import { motion } from 'framer-motion';
-import PostInfo from '../../components/PostInfo';
 
 let easing = [0.175, 0.85, 0.42, 0.96];
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/AnimationPost/[index].js]: Solved.

File: pages/_app.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+import App from 'next/app';
+import StateContext from '../components/StateContext';
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+class MyApp extends App {
+  state = {
+    manualEnabled: false,
+    isAnimationOn: true
+  };
+
+  componentDidMount = () => {
+    const manualEnabled = localStorage.getItem('manual-enabled');
+    const isAnimationOn = localStorage.getItem('animation-on');
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+      this.setState({
+        manualEnabled,
+        isAnimationOn
+      });
+      localStorage.setItem('manual-enabled', this.state.manualEnabled);
+      localStorage.setItem('animation-on', this.state.isAnimationOn);
+    }
+  };
+
+  enableManualAnimationHandler = (flag) => {
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    localStorage.setItem('manual-enabled', flag);
+    localStorage.setItem('animation-on', this.state.isAnimationOn);
+    this.setState(
+      {
+        manualEnabled: flag
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    );
+  };
+
+  toggleAnimationHandler = (event) => {
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    localStorage.setItem('animation-on', event.target.checked);
+    localStorage.setItem('manual-enabled', this.state.manualEnabled);
+    this.setState(
+      {
+        isAnimationOn: event.target.checked
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  render() {
+    const { Component, pageProps, router } = this.props;
+    return (
+      <StateContext.Provider value={{
+        manualEnabled: this.state.manualEnabled,
+        isAnimationOn: this.state.isAnimationOn,
+        enableManualAnimationHandler: this.enableManualAnimationHandler,
+        toggleAnimationHandler: this.toggleAnimationHandler}}>
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    this.setState(
+      {
+        manualEnabled: flag
+      }
+    );
+  };
+
+  toggleAnimationHandler = (event) => {
+    localStorage.setItem('animation-on', event.target.checked);
+    localStorage.setItem('manual-enabled', this.state.manualEnabled);
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+        isAnimationOn
+      });
+      localStorage.setItem('manual-enabled', this.state.manualEnabled);
+      localStorage.setItem('animation-on', this.state.isAnimationOn);
+    }
+  };
+
+  enableManualAnimationHandler = (flag) => {
+    localStorage.setItem('manual-enabled', flag);
+    localStorage.setItem('animation-on', this.state.isAnimationOn);
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+        manualEnabled: false,
+        isAnimationOn: true
+      });
+    } else {
+      this.setState({
+        manualEnabled,
+        isAnimationOn
+      });
+      localStorage.setItem('manual-enabled', this.state.manualEnabled);
+      localStorage.setItem('animation-on', this.state.isAnimationOn);
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+  componentDidMount = () => {
+    const manualEnabled = localStorage.getItem('manual-enabled');
+    const isAnimationOn = localStorage.getItem('animation-on');
+    
+    if (!manualEnabled) {
+      this.setState({
+        manualEnabled: false,
+        isAnimationOn: true
+      });
+    } else {
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+  state = {
+    manualEnabled: false,
+    isAnimationOn: true
+  };
+
+  componentDidMount = () => {
+    const manualEnabled = localStorage.getItem('manual-enabled');
+    const isAnimationOn = localStorage.getItem('animation-on');
+    
+    if (!manualEnabled) {
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: pages/_app.js [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    );
+  };
+
+  render() {
+    const { Component, pageProps, router } = this.props;
+    return (
+      <StateContext.Provider value={{
+        manualEnabled: this.state.manualEnabled,
+        isAnimationOn: this.state.isAnimationOn,
```
[the remark this answers is not in the document]
Comment (alexander-dev-hub) [reply pages/_app.js]: Solved.

File: components/Nav/index.js [hunk 22]
```diff
@@ -1,25 +1,34 @@
 
-import CheckboxWithLabel from '../CheckboxWithLabel';
+import { useContext } from 'react';
 import SwitchWithLabel from '../SwitchWithLabel';
```
Comment (anton-karlovskiy) [inline components/Nav/index.js]: line break between libraries and custom

File: components/Nav/index.js [hunk 23]
```diff
@@ -1,25 +1,34 @@
...
+import CheckboxWithLabel from '../CheckboxWithLabel';
+import AnimationEmulationContext from '../AnimationEmulationContext';
 
 const githubLink = {
   label: 'GitHub',
   href: 'https://github.com/alexander-dev-hub/react-next-motion',
```
Comment (anton-karlovskiy) [inline components/Nav/index.js]: Please change the URL to https://github.com/anton-karlovskiy/cna-memory-considerate-animation

**Pull request merged**